app/src/main/java/com/sdp/movemeet/view/navigation/Navigation.java

```diff
@@ -73,6 +73,7 @@ public static void startActivity(View view) {
     public static void goToActivityUpload(View view) {
         Context context = view.getContext();
         Intent intent = new Intent(context, UploadActivityActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
```

Comment:
How can this flag help to have a good back button behavior?

Reply:
Exactly. This prevents these activities from being recorded on the activity stack. As soon as you change the activity it is automatically destroyed. Only the MainActivity is persistent and thanks to this, on all the other pages, when we press the back button, we are brought back to the MainActivity